query-service-impl/src/test/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverterTest.java

```diff
@@ -1060,6 +1064,79 @@ public void testQueryWithLikeOperatorForResponseBodyHavingTextIndex() {
         executionContext);
   }
 
+  @Test
+  public void testQueryWithIntArrayFilter() {
+    ViewDefinition viewDefinition = getDefaultViewDefinition();
+    defaultMockingForExecutionContext();
+
+    assertPQLQuery(
+        buildSimpleQueryWithFilter(
+            createIntNotInFilter("Span.metrics.duration_millis", List.of(123))),
+        "Select span_id FROM SpanEventView WHERE "
+            + viewDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND duration_millis NOT IN (123)",
+        viewDefinition,
+        executionContext);
+  }
+
+  @Test
+  public void testQueryWithLongArrayFilter() {
+    ViewDefinition viewDefinition = getDefaultViewDefinition();
+    defaultMockingForExecutionContext();
+
+    assertPQLQuery(
+        buildSimpleQueryWithFilter(
+            createLongInFilter("Span.metrics.duration_millis", List.of(10L))),
+        "SELECT span_id FROM SpanEventView WHERE "
+            + viewDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND duration_millis IN (10)",
+        viewDefinition,
+        executionContext);
+  }
+
+  @Test
+  public void testQueryWithFloatArrayFilter() {
+    ViewDefinition viewDefinition = getDefaultViewDefinition();
+    defaultMockingForExecutionContext();
+
+    assertPQLQuery(
+        buildSimpleQueryWithFilter(
+            createFloatNotInFilter("Span.metrics.duration_millis", List.of(10.05f))),
+        "SELECT span_id FROM SpanEventView WHERE "
+            + viewDefinition.getTenantIdColumn()
+            + " = '"
+            + TENANT_ID
+            + "' "
+            + "AND duration_millis NOT IN (10.05)",
```

Comment:
Instead of reusing`duration_millis` field for all types that don't apply to it, it's cleaner to use respective fields for various types. e.g. latitude/longitude for float/double. The fields need to be just added in request_handler.conf.